rust/navigator/src/navstate.rs

```diff
@@ -610,954 +602,1307 @@ impl State {
+                                    author_info,
+                                    sufficient: sufficient.clone(),
+                                    content,
+                                };
+                                Some(ModalData::SufficientCryptoReady { f })
+                            } else {
+                                None
+                            }
                         }
+                        _ => None,
                     }
                 }
+                Modal::TypesInfo => Some(ModalData::TypesInfo {
+                    f: db_handling::interface_signer::show_types_status(dbname).unwrap(),
+                }),
                 Modal::SelectSeed => {
-                    match db_handling::interface_signer::print_all_seed_names_with_identicons(
-                        dbname,
-                        &self.seed_names,
-                    ) {
-                        Ok(a) => format!("\"seedNameCards\":{}", a),
-                        Err(e) => {
+                    match get_all_seed_names_with_identicons(dbname, &self.seed_names) {
+                        Ok(a) => Some(ModalData::SelectSeed {
+                            f: MSeeds { seed_name_cards: a },
+                        }),
+                        Err(_e) => {
                             new_navstate.alert = Alert::Error;
-                            errorline.push_str(&<Signer>::show(&e));
-                            "\"seedNameCards\":[]".to_string()
+                            Some(ModalData::SelectSeed {
+                                f: MSeeds {
+                                    seed_name_cards: vec![],
+                                },
+                            })
                         }
                     }
                 }
-                _ => "".to_string(),
+                Modal::NewSeedMenu => Some(ModalData::NewSeedMenu),
+                Modal::LogComment => Some(ModalData::LogComment),
+                Modal::NetworkDetailsMenu => Some(ModalData::NetworkDetailsMenu),
+                Modal::KeyDetailsAction => Some(ModalData::KeyDetailsAction),
+                Modal::Empty => None,
             };
 
             //Prepare alert details
             //Important! No errors could be handled in this block!
-            let alert_details = match new_navstate.alert {
-                Alert::Error => format!("\"error\":\"{}\"", errorline),
-                Alert::ErrorDisplay => format!("\"error\":\"{}\"", errorline),
-                Alert::Empty => "".to_string(),
-                Alert::Shield => "\"shield_state\":\"unknown\"".to_string(),
+            let alert_data = match new_navstate.alert {
+                Alert::Error | Alert::ErrorDisplay => Some(AlertData::ErrorData {
+                    f: format!("{{\"error\":\"{}\"}}", errorline),
+                }),
+                Alert::Empty => None,
+                Alert::Shield => match get_danger_status(dbname) {
```

Comment:
for some reason this always sends `.Past` state in swift, even when calling `get_danger_status` directly sends `false`